fastlane/lib/fastlane/actions/carthage.rb

```diff
@@ -11,8 +11,14 @@ def self.run(params)
 
         if command_name == "archive" && params[:frameworks].count > 0
           cmd.concat(params[:frameworks])
-        elsif ["update", "build", "bootstrap"].include?(command_name) && params[:dependencies].count > 0
-          cmd.concat(params[:dependencies])
+        # "update", "build" and "bootstrap" are the only commands that support "--derived-data" parameter
+        elsif ["update", "build", "bootstrap"].include?(command_name)
+          if params[:dependencies].count > 0
```

Comment:
Sorry if the previous suggestion wasn't clear - we do not need `if params[:dependencies].count > 0` anymore. 
```
elsif ["update", "build", "bootstrap"].include?(command_name)
    cmd.concat(params[:dependencies]) if params[:dependencies].count > 0
    cmd << "--derived-data #{params[:derived_data].shellescape}" if params[:derived_data]
end
```

Reply:
Oh right! My bad. Thanks. I'll take care of that.